picard/ui/options/plugins.py

```diff
@@ -246,30 +264,37 @@ def add_plugin_item(self, plugin, item=None):
         else:
             item.setText(1, plugin.version)
 
-        label = None
+        label = _("Update")
         if plugin.can_be_updated:
-            label = _("Update")
-        elif plugin.can_be_downloaded:
-            label = _("Install")
-            item.setFlags(item.flags() ^ QtCore.Qt.ItemIsUserCheckable)
-
-        if label is not None:
+            self.ui.plugins.headerItem().setText(COLUMN_NO, label)
             button = QtWidgets.QPushButton(label)
             button.setMaximumHeight(button.fontMetrics().boundingRect(label).height() + 7)
-            self.ui.plugins.setItemWidget(item, 2, button)
+            self.ui.plugins.setItemWidget(item, COLUMN_NO, button)
+
+            def download_button_process():
+                self.ui.plugins.setCurrentItem(item)
+                self.download_plugin()
+            button.released.connect(download_button_process)
+
+        if plugin.can_be_downloaded:
+            label = _("Install")
+        else :
+            label = _("Uninstall")
```

Comment:
I noticed something:
I install a plugin, then i choose to uninstall it, the dialog is shown telling me the uninstall will occur after a restart.
But the interface let the user uninstall again a plugin actually already uninstalled. The button could be replaced by "Uninstalled" or smt.

Reply:
done